### PR TITLE
Fix wrong assumption in HooksRenderer

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer.tsx [hunk 1]
```diff
@@ -33,7 +33,9 @@ export function HooksRenderer({
     return null;
   }
 
-  return object.preview.properties.map((property, index) => (
+  const objectProperties = object.preview.properties.filter(property => property.object);
```
Comment (Domiii) [inline src/ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer.tsx:36]: I feel like a utility function that filters out only Array elements from an `ObjectPreview` might be better here.

Comment (bvaughn) [reply src/ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer.tsx:36]: This change needs more description/context via an inline comment.

Comment (bvaughn) [reply src/ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer.tsx:36]: TBH I don't like this approach we're using (neither here nor the fix yesterday). 

I think we should instead be checking to see if the property is enumerable (via `.flags`)

**Review by bvaughn — CHANGES_REQUESTED** [review]

**Comment by hbenl** [comment]
> I think we should instead be checking to see if the property is enumerable (via .flags)

That would also not be correct, you can add properties to an array that are not elements (and they can be enumerable).

**Comment by bvaughn** [comment]
> > I think we should instead be checking to see if the property is enumerable (via .flags)
> 
> That would also not be correct, you can add properties to an array that are not elements (and they can be enumerable).

That's true, but it feels more semantically correct since we are iterating over properties.

Technically you could push values that aren't the shape we're expecting to the hooks array too, in which case even the number check isn't right. The thing we're guarding against with this PR isn't _unexpected data_, but unexpected protocol properties. (I think "length" used to be in `getterValues` but now is in `properties`?)

**Review by bvaughn — APPROVED** [review]
I think using the enumerable attribute seems clearer but I don't feel _super strongly_ about this, so I'm going to remove my blocking mark.

**Comment by hbenl** [comment]
> In this case, we control both ends (the thing creating/writing the array and the thing displaying it) so I don't think we need to guard against unexpected manually-defined properties.

True. But I've moved the check (`isArrayElement()`) into `utils/protocol.ts` now and since it's general-purpose I want to stick to checking whether javascript will see the property as an array element.

File: packages/replay-next/src/utils/protocol.ts [hunk 2]
```diff
@@ -437,3 +430,7 @@ export function findProtocolObjectPropertyValue<Type>(
 ): Type | null {
   return findProtocolObjectProperty(sourceObject, name)?.value ?? null;
 }
+
+export function isArrayElement(property: ProtocolProperty): boolean {
```
Comment (bvaughn) [inline packages/replay-next/src/utils/protocol.ts:434]: This name confuses me. I know this is a nit, but can you elaborate on why you chose "array element"? I would assume that meant something more like "The value contained is an array"

Comment (hbenl) [reply packages/replay-next/src/utils/protocol.ts:434]: I couldn't think of a better word for an item in an array, I usually call them array elements.

**Pull request merged**